src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -730,13 +731,13 @@ def set_last_query_line_intervals(self) -> None:
     def __copy__(self) -> "SnowflakePlan":
         if self.session._cte_optimization_enabled:
             plan = SnowflakePlan(
-                copy.deepcopy(self.queries) if self.queries else [],
+                [copy.copy(q) for q in (self.queries or [])],
```

Comment:
is `queries` a list of `Query` object or str?
i'm wondering if code change is equivalent to deepcopy..